src/components/NumberFormat/UnitNumberFormatField.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { ChangeEvent, useEffect } from 'react';
+import { TextFieldProps, TextField } from '../TextField';
+
+import { makeStyles } from '../../styles';
+import clsx from 'clsx';
+
+const useStyles = makeStyles((theme) => ({
+  textField: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+export const UnitNumberFormatField: React.FC<
+  Omit<TextFieldProps, 'onChange' | 'value'> & {
+    min?: number;
+    max?: number;
+    value: number;
+    units: string;
+    decimalScale?: number;
+    prefixUnits?: boolean;
+    onChange: (val: number) => void;
+  }
+> = (props) => {
+  const {
+    onChange,
+    onBlur,
+    value,
+    min = 0,
+    max = Number.MAX_SAFE_INTEGER,
+    units,
+    decimalScale = 0,
+    prefixUnits = false,
+    className,
+    ...other
+  } = props;
+  // formatting helper. Prefixes units if specified, otherwise suffix.
+  // also automatically adds a space if the units string is > 2 e.g $10.00 30%, 12ml, 10 slots
+  const format = (val: number) =>
+    prefixUnits
+      ? `${units}${val.toFixed(decimalScale)}`
+      : `${val.toFixed(decimalScale)}${units.length > 2 ? ' ' : ''}${units}`;
```

Comment:
Thinking forward a bit, I'm curious how this will change if we need to support RTL.  Maybe go through a provided `format` prop?  No need to change now, just something we can keep in the back of our minds 👍

Reply:
That's a good observation. There is also the case where currency symbols are not consistently on one side within LTR and RTL languages.

Reply:
Ah, nice insight on currency symbols as well.  We should support this eventually, but I think this is good to go as is if y'all don't need to support those cases just yet 👍.  We will get a lot of value out of these as is, and whoever needs to support that first (maybe Lifeology?) can add that functionality and the rest of the products benefit!